### PR TITLE
Internationalization + Japanese Language

### PR DESCRIPTION
**Pull request by mozzius**
Supercedes #412

[Crowdin](https://crowdin.com/project/graysky)

Uses Lingui for translations - still some places where translations need to be added, but looking pretty good!

@reindex-ot I've tried to copy paste your translations over - please let me know if I've made any mistakes.

### Adding/reviewing translations

Translations can be managed through Crowdin here - https://crowdin.com/project/graysky

### Running Lingui in the app

Install dependencies using `pnpm i` and run:

```
pnpm extract
```

And it will find all the translatable tasks in the project and tell you the status of the translations. Current status:

```
┌──────────┬─────────────┬─────────┐
│ Language │ Total count │ Missing │
├──────────┼─────────────┼─────────┤
│ en       │     496     │   496   │
│ ja       │     496     │    3    │
└──────────┴─────────────┴─────────┘
```

### CODE REVIEW

File: apps/expo/src/app/(tabs)/(feeds,search,notifications,self)/search/index.tsx [hunk 1]
```diff
@@ -277,11 +283,11 @@ const Suggestions = () => {
                                   {tag.name}
                                 </Text>{" "}
                                 <Text className="font-normal text-neutral-400 dark:text-neutral-500">
-                                  #{tag.tag}
+                                  <Trans>#{tag.tag}</Trans>
```
Comment (surfdude29) [inline apps/expo/src/app/(tabs)/(feeds,search,notifications,self)/search/index.tsx:286]: Sorry to jump in unasked, just to check, does this need to be marked for translation? Is it for languages where the hashtag symbol appears after the tag, something like that? 🤔

File: apps/expo/src/i18n/locales/en/messages.po [hunk 2]
```diff
@@ -0,0 +1,1732 @@
...
+msgid "Add to user list"
+msgstr ""
+
+#: src/app/composer/index.tsx:244
+msgid "Adding a description to your image makes Bluesky more accessible to people with disabilities, and helps give context to everyone. We strongly recommend adding alt text to all images."
+msgstr ""
+
+#: src/app/composer/content-warning.tsx:47
+msgid "Adult content"
+msgstr "アダルトコンテンツ"
```
Comment (surfdude29) [inline apps/expo/src/i18n/locales/en/messages.po:159]: This should just be empty I think as it's the English file

File: apps/expo/src/app/(tabs)/(feeds,search,notifications,self)/search/index.tsx (hunk 1 above)
Comment (mozzius) [reply apps/expo/src/app/(tabs)/(feeds,search,notifications,self)/search/index.tsx:286]: Yeah I thought there was a small chance a translator might wanna add some additional context here

File: apps/expo/src/i18n/locales/en/messages.po (hunk 2 above)
Comment (mozzius) [reply apps/expo/src/i18n/locales/en/messages.po:159]: Whoops yeah good catch

File: apps/expo/src/components/feed-post.tsx [hunk 3]
```diff
@@ -246,54 +247,38 @@ const FeedPostInner = ({
...
-                })()
+              ? AppBskyFeedDefs.isPostView(item.reply.parent) && (
+                  <Link
+                    href={path(
+                      `/profile/${
+                        item.reply.parent.author.did
+                      }/post/${item.reply.parent.uri.split("/").pop()}`,
+                    )}
+                    asChild
+                    accessibilityHint="Opens parent post"
```
Comment (surfdude29) [inline apps/expo/src/components/feed-post.tsx:258]: Wow this is turning into quite a huge PR, nice job! 👏

Sorry to be that guy again, but I did notice this `accessibilityHint` that I think should probably be marked for translation?

**Comment by reindex-ot** [comment]
hi, i18n supported?

**Comment by mozzius** [comment]
> hi, i18n supported?

Yes, translation files are kept in `apps/expo/src/i18n/locales` (I think you saw already)

**Comment by mozzius** [comment]
I have created a crowdin project to make this easier:

https://crowdin.com/project/graysky

**Comment by surfdude29** [comment]
> I have created a crowdin project to make this easier:
> 
> https://crowdin.com/project/graysky

That's what the Ukrainian translation team for Bluesky uses, and pfrazee said it's on his TODOs to look into using it (sorry if I'm telling you stuff you already know :)

File: apps/expo/src/components/handle-availability.tsx [hunk 4]
```diff
@@ -53,21 +57,27 @@ export const HandleAvailabilityResult = ({
...
+            </Text>
           </View>
         );
       case "taken":
         return (
           <View className="flex-row items-center">
             <XCircleIcon className="mr-1.5 text-red-500" size={14} />
-            <Text className="text-sm text-red-500">Handle is taken</Text>
+            <Text className="text-sm text-red-500">
+              <Trans>Handle is taken</Trans>
```
Comment (surfdude29) [inline apps/expo/src/components/handle-availability.tsx:70]: What about changing this to "Handle already taken" to match one of the strings in change-handle.tsx?

File: apps/expo/src/components/screens/change-password-flow.tsx [hunk 5]
```diff
@@ -66,15 +69,16 @@ export const ChangePasswordFlow = ({ defaultEmail = "" }: Props) => {
...
+          message: _(msg`Please try again`),
           status: "danger",
         });
       } else {
         console.error(err);
         showToastable({
-          title: "An error occured",
-          message:
-            "Please try again - are you sure your reset code is correct?",
+          title: _(msg`An error occured`),
```
Comment (surfdude29) [inline apps/expo/src/components/screens/change-password-flow.tsx:78]: Typo here, "occurred" is missing an r

**Comment by mozzius** [comment]
I've added some outstanding things that need translation

**Comment by mozzius** [comment]
Almost done! Just found a couple more untranslated strings

File: apps/expo/src/components/screens/profile/profile-info.tsx
Comment (surfdude29) [inline apps/expo/src/components/screens/profile/profile-info.tsx:1]: I think the strings here for the profile context menu should be marked for translation: L164-169

**Pull request merged**